index.js

```diff
@@ -231,4 +231,55 @@ transform.resetTypes = function resetTypes() {
   registerTypes = {};
 };
 
-module.exports = transform;
+function addPermissions(permissions, key, type) {
+  return permissions[key] && permissions[key][type] !== undefined ? permissions[key][type] : permissions[type]
+}
+
+function getFormOptions(properties, fieldsData = {}, permissions, newOptions = {}, first = false) {
+  Object.entries(properties).forEach((data) => {
+      const [key, value] = data
+      if (first) {
```

Comment:
You don't need a special case for the first iteration, the top level data is just a struct. You can tell the difference between struct/list for recursion conditional branching by checking for the key "fields"/"items" in the permission json.